nf_core/components/nfcore_component.py

```diff
@@ -217,8 +217,10 @@ def get_inputs_from_main_nf(self) -> None:
                         input_val = match.group(4).split(",")[0]  # handle `files, stageAs: "inputs/*"` cases
                     if input_val:
                         channel_elements.append({input_val: {}})
-                if len(channel_elements) > 0:
+                if len(channel_elements) > 1:
                     inputs.append(channel_elements)
+                elif len(channel_elements) > 0:
```

Comment:
```suggestion
                elif len(channel_elements) == 0:
```

Reply:
Would be `== 1` if we want to use the equal 😄 
```suggestion
                elif len(channel_elements) == 1:
```